core/domain/html_cleaner.py

```diff
@@ -458,13 +460,127 @@ def convert_to_text_angular(html_data):
     return unicode(soup).replace('<br/>', '<br>')
 
 
-def convert_tag_contents_to_text_angular(html_data):
+def convert_to_ck_editor(html_data):
+    """This function converts the html to CKEditor supported format.
+
+    Args:
+        html_data: str. HTML string to be converted.
+
+    Returns:
+        str. The converted HTML string.
+    """
+    if not len(html_data):
+        return html_data
+
+    # <br> is replaced with <br/> before conversion because BeautifulSoup
+    # in some cases adds </br> closing tag and br is reported as parent
+    # of other tags which produces issues in migration.
+    html_data = html_data.replace('<br>', '<br/>')
+
+    # To convert the rich text content within tabs and collapsible components
+    # to valid CKEditor format. If there is no tabs or collapsible component
+    # convert_tag_contents_to_rte_format will make no change to html_data.
+    html_data = convert_tag_contents_to_rte_format(
+        html_data, feconf.RTE_FORMAT_CKEDITOR)
+
+    soup = bs4.BeautifulSoup(html_data.encode('utf-8'), 'html.parser')
+
+    # Replaces b tag with strong tag.
+    for b in soup.findAll('b'):
+        b.name = 'strong'
+
+    # Replaces i tag with em tag.
+    for i in soup.findAll('i'):
+        i.name = 'em'
+
+    # This block wraps p tag in li tag if the parent of p is ol/ul tag. Also,
+    # if the parent of p tag is pre tag, it unwraps the p tag.
+    for p in soup.findAll('p'):
+        if p.parent.name == 'pre':
+            p.unwrap()
+        elif p.parent.name in ['ol', 'ul']:
+            p.wrap(soup.new_tag('li'))
+
+    # Replaces <p><br><p> with <p>&nbsp;</p>.
+    for br in soup.findAll('br'):
+        parent_tag = br.parent
+        if parent_tag.name == 'p' and parent_tag.get_text() == '':
+            br.replaceWith('&nbsp;')
+
+    # In CKEditor format ol/ul cannot be the parent of ol/ul. So this block
```

Comment:
You're saying this does

    <ol><ol>...</ol></ul> --> <ol><li><ol>...</ol></li></ol>

If so, I would modify the second sentence to say `So if this block finds an ol/ul tag that is the child of another ol/ul tag, it will wrap the inner ol/ul tag into an li tag.`

Or better yet, provide an actual concrete example (like I did above).

Reply:
Done, PTAL!